src/P3_particle_properties.jl

```diff
@@ -1,333 +1,498 @@
+# TODO: Implement `F_liq` as another P3State-like struct
 """
-    α_va_si(p3)
+    P3State{FT}
 
- - p3 - a struct with P3 scheme parameters
+State of the P3 scheme.
 
-Returns `α_va` coefficient for the assumed particle mass(size) relation for
-large unrimed ice and dense nonspherical ice, in base SI units: kg m^(-β_va).
-`β_va` is another coefficient of the mass(size) relation.
-From measurements of mass grown by vapor diffusion and aggregation
-in midlatitude cirrus by Brown and Francis (1995)
-doi: 10.1175/1520-0426(1995)012<0410:IMOTIW>2.0.CO;2
+This struct bundles the P3 parameterizations `params`, the provided rime state (`F_rim`, `ρ_r`), 
+    and the derived threshold variables (`D_th`, `D_gr`, `D_cr`, `ρ_g`).
+
+# Fields
+$(FIELDS)
+
+# Examples
+
+To obtain a `P3State` object, use the [`get_state`](@ref) function.
+
+```julia
+FT = Float64
+params = ParametersP3(FT)
+state = get_state(params; F_rim = FT(0.5), ρ_r = FT(916.7))
+```
 """
-α_va_si(p3::PSP3{FT}) where {FT} = p3.α_va * 10^(6 * p3.β_va - 3)
+Base.@kwdef struct P3State{FT}
+    "[`CMP.ParametersP3`](@ref) object"
+    params::PSP3{FT}
+
+    "Rime mass fraction"
+    F_rim::FT
+    "Rime density"
+    ρ_r::FT
+    "Graupel density"
+    ρ_g::FT
+
+    "Spherical ice threshold"
+    D_th::FT
+    "Graupel threshold"
+    D_gr::FT
+    "Partially rimed ice threshold"
+    D_cr::FT
+end
 
 """
-    D_th_helper(p3)
+    get_state(params; F_rim, ρ_r)
+
+Create a [`P3State`](@ref) object from a [`CMP.ParametersP3`](@ref) object and rime state parameters.
+
+# Arguments
+- `params`: [`CMP.ParametersP3`](@ref) object
+- `F_rim`: rime mass fraction
+- `ρ_r`: rime density
 
- - p3 - a struct with P3 scheme parameters
+!!! note "Floating point type"
+    The floating point type of `params`, `F_rim`, and `ρ_r` must be the same.
 
-Returns the critical size separating spherical and nonspherical ice, in meters.
-Eq. 8 in Morrison and Milbrandt (2015).
+# Examples
+
+```julia
+FT = Float64
+params = ParametersP3(FT)
+state = get_state(params; F_rim = FT(0.5), ρ_r = FT(916.7))
+```
 """
-D_th_helper(p3::PSP3{FT}) where {FT} =
-    (FT(π) * p3.ρ_i / 6 / α_va_si(p3))^(1 / (p3.β_va - 3))
+function get_state(params::PSP3{FT}; F_rim, ρ_r) where {FT}
```

Comment:
Thinking about coupling with the Atmos model I would suggest to avoid caching too much. So rather than precomputing `F_rim` and passing it around, it is better to compute on the fly. 

Now that I think about this, I wonder if we even want to cache the thresholds into the struct. I used it do this because they were solved for rootsolver. Bu now that we have analytical formulae for them, it might be more efficient to recompute them when needed